test/variable.jl

```diff
@@ -500,6 +500,19 @@ function test_variables_constrained_on_creation(ModelType)
     @test num_constraints(model, typeof(x), MOI.PositiveSemidefiniteConeTriangle) == 1
 end
 
+function test_batch_delete_variables(ModelType)
+    model = ModelType()
+    @variable(model, x[1:3] >= 1)
+    @objective(model, Min, sum([1, 2, 3] .* x))
+    @test all(is_valid.(model, x))
+    delete(model, x[[1, 3]])
+    @test all((!is_valid).(model, x[[1, 3]]))
+    @test is_valid(model, x[2])
+    second_model = ModelType()
+    @test_throws Exception JuMP.delete(second_model, x[2])
```

Comment:
We usually have more specific tests that include the error messages. See examples above.

Reply:
I see that `test_variable_is_valid_delete` doesn't do any better, so feel free to ignore this advice.

Reply:
The problem with this approach is that the current JuMPExtension does not throw the same error message, nor the same specific type of exception. For the single variable deletion (that was already there and I did not touch), [it is an error in the JuMP](https://github.com/JuliaOpt/JuMP.jl/blob/master/src/variables.jl#L173) and [an assertion in JuMPExtension](https://github.com/JuliaOpt/JuMP.jl/blob/master/test/JuMPExtension.jl#L74). I think its is not the place of this PR to fix this.

Reply:
For now, I will leave this out of this PR.